### ISSUE TITLE
Fails to connect and update firmware v1.3.4

### ISSUE THREAD

**Issue by gvolpe** [Dygmalab/Bazecor#525]
**Describe the bug**
Dygma Defy is detected, but only connected in "bootloader mode". If I try to update the firmware, I only get a blank screen and the entire program just hangs (I tried leaving it for more than one hour with the same result). It happens both when I connect via USB and in wireless mode (RF).

There's also a strange behavior you can observe in the video linked below: if I click on the firmware tab and then I got back to the connection option, it shows my Dygma is connected but I can't do anything with it. 

I can only see this error when I run the program, but I'm not sure whether it's a critical one or not:

```console
Electron's autoUpdater does not support the 'linux' platform
CHECKING BACKUP FOLDER VALUE /home/gvolpe/Dygma/Backups
[10710:1009/212809.422364:ERROR:object_proxy.cc(590)] Failed to call method: org.freedesktop.portal.Settings.Read: object_path= /org/freedesktop/portal/desktop: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.portal.Desktop was not provided by any .service files
```

Are there any logs produced by the application anywhere else?

**To Reproduce**
Not sure how you can reproduce outside of Linux, but I've made a video: https://youtu.be/CG9dtL3SwX8

**Expected behavior**
It should connect to my Defy and allow me to make changes.

**Screenshots**
If possible, please also include a log from the "developer console". You can access it by pressing `Control + Shift + I`. When it appears, right click inside it, and select "Save as", and attach the saved file to this report.

This is currently broken. Why does Bazecor need to access a root directory?

```console
[10921:1009/212825.192021:ERROR:platform_shared_memory_region_posix.cc(214)] Creating shared memory in /dev/shm/.org.chromium.Chromium.p6BOrH failed: No such file or directory (2)
[10921:1009/212825.192166:ERROR:platform_shared_memory_region_posix.cc(217)] Unable to access(W_OK|X_OK) /dev/shm: No such file or directory (2)
[10921:1009/212825.192208:FATAL:platform_shared_memory_region_posix.cc(219)] This is frequently caused by incorrect permissions on /dev/shm.  Try 'sudo chmod 1777 /dev/shm' to fix.
```

**Desktop (please complete the following information):**
 - OS: NixOS 23.11.20230929.f5892dd (Tapir) x86_64
 - Bazecor Version: 1.3.4

**Additional context**
I've received my Defy today and would really love to start using it, but it's impossible without configuring the thumb clusters at least :sob:

**Comment by gvolpe** [comment]
I've managed to get to the firmware update screen by following a specific sequence of actions (found out only by tinkering around), but now I'm seeing an error message saying I should connect both sides of the keyboard... except they are already connected 🤷🏽‍♂️ 

I also tried different ports directly on my laptop without luck. Moreover, I've watched [this troubleshooting video](https://www.youtube.com/watch?v=-93k5B5quI8), but it doesn't really explain much about the issue I'm having.

Here's another recording I've made that shows what's going on: https://youtu.be/1GqnTdfn_V0

**Comment by jacott** [comment]
You can't update firmware on a virtual device so you second video is irrelevant. 

You are being prompted to install to udev rules every time? Looks like that is not working.  You should have a file like:

```
$ cat /etc/udev/rules.d/10-dygma.rules 
SUBSYSTEMS=="usb", ATTRS{idVendor}=="1209", ATTRS{idProduct}=="2201", GROUP="users", MODE="0666"
SUBSYSTEMS=="usb", ATTRS{idVendor}=="1209", ATTRS{idProduct}=="2200", GROUP="users", MODE="0666"
SUBSYSTEMS=="usb", ATTRS{idVendor}=="35EF", MODE="0666"
$
```

You should only try to update the firmware in wired mode with the neuron plugged directly into the computer USB 3 port; don't use RF or bluetooth. Also with the keeb halves turned off.

Try plugging in the neuron whilst holding down the escape (top-left) key. Also hold key during flashing if you get that far.

I get the same shared memory FATAL error you do but the permissions are correct.

Disclaimer: I have not yet received my Defy so just mostly basing this off what I have read on discord.

**Comment by gvolpe** [comment]
> You can't update firmware on a virtual device so you second video is irrelevant.

Oh that's what it is? It's not very intuitive that's happening, as my keyboard *is* connected.

> You are being prompted to install to udev rules every time? Looks like that is not working.

That's because of [this other issue](https://github.com/Dygmalab/Bazecor/issues/370) I reported long ago that's still unsolved, but I do have the udev rules and it's working for users of the Dygma Raise, for instance. See the Nix-packaged version we're using here: https://github.com/NixOS/nixpkgs/pull/254264

These are the udev rules I have set:

```ini
SUBSYSTEMS=="usb", ATTRS{idVendor}=="1209", ATTRS{idProduct}=="2201", GROUP="users", MODE="0666"
SUBSYSTEMS=="usb", ATTRS{idVendor}=="1209", ATTRS{idProduct}=="2200", GROUP="users", MODE="0666"
SUBSYSTEMS=="usb", ATTRS{idVendor}=="35EF", MODE="0666"
```

> Try plugging in the neuron whilst holding down the escape (top-left) key. Also hold key during flashing if you get that far.

I just get the same blank screen I reported initially (first video).

**Comment by gvolpe** [comment]
My user is part of the `uucp` group as well, as suggested on [this article](https://support.dygma.com/hc/en-us/articles/360016653338-Fixing-the-Insufficient-Permission-Error-in-Bazecor-on-Linux-Dygma), still no luck.

```console
$ groups gvolpe
gvolpe : users wheel uucp lp networkmanager scanner docker vboxusers
```

**Comment by jacott** [comment]
for me on Ubuntu:

```
$ groups geoffj
geoffj : geoffj adm cdrom sudo dip plugdev kvm lpadmin lxd sambashare docker libvirt
```

plugdev might be important 🤷

**Comment by gvolpe** [comment]
I started using their latest release published yesterday, and this error is now gone:

```
[10710:1009/212809.422364:ERROR:object_proxy.cc(590)] Failed to call method: org.freedesktop.portal.Settings.Read: object_path= /org/freedesktop/portal/desktop: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.portal.Desktop was not provided by any .service files
```

I also managed to fix the connection issues by using the following udev rules instead:

```
SUBSYSTEMS=="usb", ATTRS{idVendor}=="35ef", ATTRS{idProduct}=="0012", GROUP="users", MODE="0666"
```

So I can now use my Defy 🥳! I didn't get to test the firmware update process, though, as mine's up-to-date. I'll close this issue for now, and I'll re-open if there are any firmware update issues in the future (or raise a new issue).

--- 

NOTE to the developers @javierguzman @AlexDygma: Github releases are supposed to be immutable. Uploading binaries that don't correspond to the git tag is a terrible practice, and it causes all sorts of issues with package managers and the github cache. Please consider making a new patch release next time instead.

In the meantime, I've uploaded the new AppImage to a new repo for personal use: https://github.com/gvolpe/bazecor-1.3.4-bug-fix/releases/tag/v1.3.4-hot-fix

For the Nix users out there, here's how I packaged it: https://github.com/gvolpe/nix-config/commit/4a9c230298e435c92cf4e795d711cddfb66996b2
